sdk/veracruz-app-lib/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE_MIT.markdown` file in the Veracruz root directory
+//! for information on licensing and copyright.
+
+use std::fs::File;
+use std::os::wasi::io::{FromRawFd, RawFd};
+
+mod veracruz_si_import {
+    #[link(wasm_import_module = "veracruz_si")]
+    extern "C" {
+        pub fn fd_create(x: u32) -> u32;
+    }
+}
+
+mod veracruz_si {
+    pub fn fd_create(fd: *mut crate::RawFd) -> u32 {
+        unsafe { crate::veracruz_si_import::fd_create(fd as u32) }
+    }
+}
+
+pub fn fd_create() -> std::io::Result<File> {
```

Comment:
Can you add a `rustdoc` comment describing what `fd_create` does and its failure modes?

Reply:
Done.